include/nbl/video/utilities/CComputeBlit.h

```diff
@@ -0,0 +1,900 @@
+		const uint32_t alphaBinCount = DefaultAlphaBinCount);
+
+	core::smart_refctd_ptr<video::IGPUComputePipeline> getNormalizationPipeline(const asset::IImage::E_TYPE imageType, const asset::E_FORMAT outFormat,
+		const uint32_t alphaBinCount = DefaultAlphaBinCount)
+	{
+		const auto workgroupDims = getDefaultWorkgroupDims(imageType);
+		const uint32_t paddedAlphaBinCount = getPaddedAlphaBinCount(workgroupDims, alphaBinCount);
+		const SNormalizationCacheKey key = { imageType, paddedAlphaBinCount, outFormat };
+
+		if (m_normalizationPipelines.find(key) == m_normalizationPipelines.end())
+		{
+			auto specShader = createNormalizationSpecializedShader(imageType, outFormat, paddedAlphaBinCount);
+			m_normalizationPipelines[key] = m_device->createComputePipeline(nullptr, core::smart_refctd_ptr(m_blitPipelineLayout[EBT_COVERAGE_ADJUSTMENT]), std::move(specShader));
+		}
+
+		return m_normalizationPipelines[key];
+	}
+
+	template <typename KernelX, typename KernelY, typename KernelZ>
+	core::smart_refctd_ptr<video::IGPUSpecializedShader> createBlitSpecializedShader(
+		const asset::E_FORMAT inFormat,
+		const asset::E_FORMAT outFormat,
+		const asset::IImage::E_TYPE imageType,
+		const core::vectorSIMDu32& inExtent,
+		const core::vectorSIMDu32& outExtent,
+		const asset::IBlitUtilities::E_ALPHA_SEMANTIC alphaSemantic,
+		const KernelX& kernelX, const KernelY& kernelY, const KernelZ& kernelZ,
+		const uint32_t workgroupSize = DefaultBlitWorkgroupSize,
+		const uint32_t alphaBinCount = DefaultAlphaBinCount)
+	{
+		const auto workgroupDims = getDefaultWorkgroupDims(imageType);
+		const auto paddedAlphaBinCount = getPaddedAlphaBinCount(workgroupDims, alphaBinCount);
+
+		using blit_utils_t = asset::CBlitUtilities<KernelX, KernelY, KernelZ>;
+
+		const auto scaledKernelX = blit_utils_t::constructScaledKernel(kernelX, inExtent, outExtent);
+		const auto scaledKernelY = blit_utils_t::constructScaledKernel(kernelY, inExtent, outExtent);
+		const auto scaledKernelZ = blit_utils_t::constructScaledKernel(kernelZ, inExtent, outExtent);
+
+		std::ostringstream shaderSourceStream;
+		shaderSourceStream
+			<< "#version 460 core\n"
+			<< "#define _NBL_GLSL_WORKGROUP_SIZE_X_ " << workgroupSize << "\n"
+			<< "#define _NBL_GLSL_WORKGROUP_SIZE_Y_ " << 1 << "\n"
+			<< "#define _NBL_GLSL_WORKGROUP_SIZE_Z_ " << 1 << "\n"
+			<< "#define _NBL_GLSL_BLIT_DIM_COUNT_ " << static_cast<uint32_t>(imageType) + 1 << "\n"
+			<< "#define _NBL_GLSL_BLIT_ALPHA_BIN_COUNT_ " << paddedAlphaBinCount << "\n";
+
+		const auto castedFormat = getOutImageViewFormat(outFormat);
+		const uint32_t outChannelCount = asset::getFormatChannelCount(outFormat);
+		// Todo(achal): All this belongs in validation
+		{
+			const uint32_t inChannelCount = asset::getFormatChannelCount(inFormat);
+			assert(outChannelCount <= inChannelCount);
+
+			// inFormat should support SAMPLED_BIT format feature
+		}
```

Comment:
you don't need this, and you don't need `inFormat` argument, the blit only cares about the number of output channels

Reply:
Resolved.